app/Http/Controllers/API/MaterialApiController.php

```diff
@@ -27,8 +27,8 @@ public function index()
 
     public function getByCategory($categoryId){
         //return all data of materials by category id
-        $categories = Category::find($categoryId);
-        return $categories->materials ;
+        $categories = Category::find($categoryId)->materials->skip(0)->take(10);
```

Comment:
for me, I would use Material model with [whereHas](https://laravel.com/docs/8.x/eloquent-relationships#querying-relationship-existence)